### PR TITLE
Reorganize `@aws-amplify/ui` TS types into multiple files

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

Our `@aws-amplify/ui` TS types have been becoming monolithic, with all the features we've been adding. This PR separates it into multiple types to help future feature works. 

**Goals**:
- Split types per use case (basic types vs cognito types vs state machine types)
- Ensure types get imported without circular dependencies.

**Organization**

image

(arrow denotes denotes type dependency)

#### Description of how you validated changes

CI workflow

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/types/authenticator/attributes.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html
+export const signUpFieldsWithDefault = [
```
Comment (reesscot) [inline packages/ui/src/types/authenticator/attributes.ts:2]: Could this be broken out further into a signUpFields file? Half this file is just SignUp related types.

File: packages/ui/src/types/authenticator/form.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+export type FormFields = {
+  [key in formFieldComponents]?: formField;
+};
+export interface formField {
+  [key: string]: formFieldTypes;
+}
+
+export interface formFieldTypes {
+  labelHidden?: boolean;
```
Comment (reesscot) [inline packages/ui/src/types/authenticator/form.ts:22]: I know this is unrelated to your refactor, but shouldn't these have JSDoc comments for intellisense?

File: packages/ui/src/types/authenticator/stateMachine/event.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  | 'RESET_PASSWORD'
+  | 'SIGN_IN'
+  | 'SIGN_OUT'
+  | 'SIGN_UP'
+  | 'SKIP'
+  | 'SUBMIT'
+  | 'INIT'
+  | InvokeActorEventTypes;
+
+export type AuthEventData = Record<PropertyKey, any>; // TODO: this should be typed further
```
Comment (reesscot) [inline packages/ui/src/types/authenticator/stateMachine/event.ts:21]: Any chance we can get do this TODO?

**Review by reesscot — APPROVED** [review]
This is great, thanks for organizing these types. A couple non-blocking comments below.

File: packages/ui/src/types/authenticator/stateMachine/event.ts (hunk 3 above)
Comment (wlee221) [reply packages/ui/src/types/authenticator/stateMachine/event.ts:21]: https://github.com/aws-amplify/amplify-ui/pull/1386 ;)

File: packages/ui/src/types/authenticator/form.ts (hunk 2 above)
Comment (wlee221) [reply packages/ui/src/types/authenticator/form.ts:22]: Yep, we should especially for public ones. Let me add some while I'm add it.

File: packages/ui/src/types/authenticator/attributes.ts (hunk 1 above)
Comment (wlee221) [reply packages/ui/src/types/authenticator/attributes.ts:2]: Good point. To be fair they're misnomers --  these fields are applicable to other screens like `ForgotNewPassword` and not just `SignUp` (#1155, #1176). 

I'm doing a form refactor today, so I'll re-address it there.

File: packages/ui/src/types/authenticator/form.ts (hunk 2 above)
Comment (wlee221) [reply packages/ui/src/types/authenticator/form.ts:39]: Done! https://github.com/aws-amplify/amplify-ui/pull/1456/commits/977acc65575398ba1c7d337f66f2023b11a1102b

**Comment by wlee221** [comment]
Tried to add some type name casing changes here https://github.com/aws-amplify/amplify-ui/pull/1456/commits/16019a8a6ae3602bc1b66717db9003404935e240, but this required more file changes so punting it to another PR.

Only change after your review is jsdocs @reesscot, I'll go ahead and merge this PR.

**Pull request merged**